src/tools/csg_imc_solve.cc

```diff
@@ -104,4 +107,19 @@ void CG_IMC_solve::Run() {
   x.y() = -inverse * A.transpose() * B.y();
 
   x.Save(outputfile);
+
+  if (OptionsMap().count("idxfile")) {
+    std::string idxfile = _op_vm["idxfile"].as<std::string>();
+    std::vector<std::pair<std::string, votca::tools::RangeParser> > ranges =
+        votca::csg::imcio_read_index(idxfile);
+    votca::tools::Table tbl_in;
+    tbl_in.Load(outputfile);
```

Comment:
@JensWehner maybe you can thing about a better solution here than reading it in again.

Reply:
Like this?

Reply:
Superb!